src/stories/Button.tsx

```diff
@@ -0,0 +1,50 @@
+import React, { useMemo } from "react";
+
+const getVariantClasses = (variant: string) => {
+  switch (variant) {
+    case "filled": {
+      return "text-white bg-purple-800 border-purple-800 border-2";
+    }
+    case "outlined": {
+      return "text-purple-800 bg-transparent border-slate-400 border-2";
+    }
+    case "text": {
+      return "text-purple-800 bg-transparent border-transparent border-2";
+    }
+    case "tonal": {
+      return "text-slate-800 bg-purple-200 border-transparent border-2";
+    }
+    case "elevated": {
+      return "text-purple-800 bg-slate-100 border-grey-200 border-2 shadow-md";
+    }
+    default: {
+      return "text-white bg-purple-800 border-purple-800 border-2";
+    }
+  }
+};
```

Comment:
This is a nitpick (nit) so feel free to take it or leave it, but you could simplify this markup a bit by using a package like [classnames](https://www.npmjs.com/package/classnames).

It would look something like this...


```jsx
function setButtonStyles(variant: string) {
  return cx(
    // base styles
    'cursor-pointer rounded-full px-4 py-2',
    {
      // handle variants
      // filled
      'text-white bg-purple-800 border-purple-800 border-2':
        variant === 'filled',
      // outlined
      'text-purple-800 bg-transparent border-slate-400 border-2':
        variant === 'outlined',
      // text
      'text-purple-800 bg-transparent border-transparent border-2':
        variant === 'text',
      // tonal
      'text-slate-800 bg-purple-200 border-transparent border-2':
        variant === 'tonal',
      // elevated
      'text-purple-800 bg-slate-100 border-grey-200 border-2 shadow-md':
        variant === 'elevated',
    },
  );
}

export const Button = ({
  variant = "filled",
  label = "Button",
  // showIcon = false,
  // icon = null,
  ...props
}) => {
  ...

  return (
    <button
      type="button"
      className={`${setButtonStyles(variant)}`}
      {...props}
    >
      {label}
    </button>
  );
};
```

Reply:
That makes sense, thanks for the suggestion! I'll try switching to this method since it looks more readable.